docs/source/howto/run_containerized_codes.rst

```diff
@@ -0,0 +1,151 @@
+.. _how-to:run-containerized_codes:
+
+****************************************
+How to setup and run containerized codes
+****************************************
+
+This how-to walks you through the steps of setting up a compute resource, setting up containerized codes on that computer, and submitting a calculation through AiiDA (with more details supplement to :ref:`introductory tutorial <tutorial:basic:calcjob>`).
+
+.. important::
+
+    Running code in the container usually require mapping the current directory (where the input files locate) to the working directory in the container.
+    Current directory is specified in the job script as ``$PWD``.
+    The computer setup to run containerized code **MUST** set ``use_double_quotes`` to ``True`` to escape the command line parameters using double quotes.
+
+There are two type of codes the ``installed`` and ``portable``, which are all supported to be run in the container.
+
+We use `singularity`_ as the containerized technology in this section.
+Singularity is easy to install and designed to supported the scientific computiong in HPC workd.
+There are other similar containerized technologies for HPC such as `sarus`_ and `shifter`_ are supposed to be support by this feature.
+We have Sarus and Singularity tested to run with this containerized code setup.
+
+Installed containerized code
+============================
+
```

Comment:
We should add entries for the two new plugins to the table in `docs/source/topics/data_types.rst` below the `InstalledCode` and `PortableCode`.

Reply:
True, added. But the Markdown table is not well indented, do I need to do it by hand or is there an online tool for it?

Reply:
I am afraid I also did it by hand. I am sure there must be a tool for it, but I think quickly adjusting manually is going to be faster than finding the tool and configuring it for this use case...

Reply:
Restructured text really sucks for formatting like this

Reply:
No worries, I already did it by hand. Not to much pain.